library/persistence.js

```diff
@@ -0,0 +1,138 @@
+const { MongoClient } = require('mongodb')
+
+function connect (url = 'mongodb://localhost:27017', dbName = 'pinano', options = {}) {
+  options.useNewUrlParser = true
+  return MongoClient.connect(url, options).then(client => {
+    return new MongoManager(client, dbName)
+  })
+}
+
+class MongoManager {
+  constructor (client, dbName) {
+    this.client = client
+    this.dbName = dbName
+    this.db = this.client.db(this.dbName)
+  }
+
+  dropDatabase () {
+    return this.db.dropDatabase()
+  }
+
+  newUserRepository () {
+    return new MongoUserRepository({ userCollection: this.db.collection('users') })
+  }
+
+  newGuildRepository () {
+    return new MongoGuildRepository({ guildCollection: this.db.collection('guilds') })
+  }
+
+  async shutdown () {
+    await this.client.close()
+  }
+}
+
+function makeUser (userId) {
```

Comment:
This is non-blocking, but it feels slightly weird to me that the ```client``` object has two notions of what a User is: the standard Discord notion of a User, and the metadata we're storing to track practice times for the former (ditto for Guild). I'm not sure if this is just a matter of naming or it makes sense for this function to be part of the repository (static?).

Just musing out loud here.

Reply:
Hmmm yes, it could be confusing. We may need to rename some things for clarity later...